src/NSwag.CodeGeneration.CSharp/Models/CSharpOperationModel.cs

```diff
@@ -248,11 +248,30 @@ protected override string ResolveParameterType(OpenApiParameter parameter)
 
             if (schema.IsBinary)
             {
+                bool useFormFile = false;
+                if (_settings is CSharpControllerGeneratorSettings controllerSettings)
+                {
+                    if (controllerSettings.ControllerTarget == CSharpControllerTarget.AspNetCore)
+                    {
+                        useFormFile = true;
+                    }
+                }
+
                 if (parameter.CollectionFormat == OpenApiParameterCollectionFormat.Multi && !schema.Type.HasFlag(JsonObjectType.Array))
                 {
+                    if (useFormFile)
+                    {
+                        return "System.Collections.Generic.IEnumerable<Microsoft.AspNetCore.Http.IFormFile>";
+                    }
+
                     return "System.Collections.Generic.IEnumerable<FileParameter>";
```

Comment:
Sorry for my late reply. 

I think one of the reasons to use FileParameter (also for controllers) is that the generated service class is independent of the technology (ASP.NET Core) and thus should use FileParameter but then this has to be transformed to a IFormFile when calling the generated methods from the generated controller. Not sure whether it's worth the effort though (and I'm personally not using controller generation). 

What do you think?

Reply:
I don't think that it's worth the effort to map FileParameter to IFormFile in this case, given that we're able to confirm that we're targeting ASP.Net core and it is the accepted way to handle multi-part form submissions with attached files. Is this something that we could keep moving forward?